types/validatingwh_config.go

```diff
@@ -0,0 +1,95 @@
+package types
+
+import (
+	"strings"
+
+	"github.com/koki/json"
+	serrors "github.com/koki/structurederrors"
+	"k8s.io/api/admissionregistration/v1beta1"
+)
+
+type ValidatingWebhookConfigWrapper struct {
+	ValidatingWebhookConfig ValidatingWebhookConfig `json:"validatingwh_config"`
+}
+
+type ValidatingWebhookConfig struct {
+	Version     string            `json:"version,omitempty"`
+	Cluster     string            `json:"cluster,omitempty"`
+	Name        string            `json:"name,omitempty"`
+	Namespace   string            `json:"namespace,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	Rules map[string][]ValidatingWebhookRuleWithOperations `json:"rules,omitempty"`
+}
+
+type ValidatingWebhookRule struct {
```

Comment:
Is this used anywhere?

Reply:
I see where this is being used. I would rather use one struct like this

type ValidatingWebhookRule struct{
  Groups         []string                                `json:"groups,omitempty"` 
  Versions       []string                                `json:"versions,omitempty"`
  Resources    []string                                `json:"resources,omitempty"`
  Operations   []v1beta1.OperationType    `json:"operations,omitempty"`
}

This way changes need not be made in two places if Groups, Versions of Resources fields needs to be updated.